torch/_inductor/fx_passes/fuse_attention.py

```diff
@@ -141,6 +146,44 @@ def _sfdp_replacement_6(query, key, value, attn_mask, dropout_p):
 # TODO(jansel): make these pattern work with lowmem_dropout=True
 
 
+def _sfdp_scale_factor_check(scale_factor_op):
+    def fn(match):
+        # make sure the mul(div) for the scale factor is scalar mul(div).
+        # bmm->view->mul(div)
+        matmuls = filter_nodes(match.nodes, aten.bmm)
+        if len(matmuls) < 2:
```

Comment:
Shouldn't this already be checked by the pattern?  Can you give an example and add a test for when this fails?

Reply:
Yes, this has been checked by the pattern, I do that to make sure it is all right. I will remove it to simplify it.